storage/src/translator.rs

```diff
@@ -120,6 +123,69 @@ define_cap_translator!(TwoCap, 2, u16);
 define_cap_translator!(FourCap, 4, u32);
 define_cap_translator!(EightCap, 8, u64);
 
+/// Define a special array type for which we'll implement our own identity hasher. This avoids the
+/// overhead of the default Array hasher which unnecessarily (for our use case) includes a length
+/// prefix.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct UnhashedArray<const N: usize> {
+    pub inner: [u8; N],
+}
+
+impl<const N: usize> Hash for UnhashedArray<N> {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        const { assert!(N <= 8 && N > 0) };
+        state.write(&self.inner);
+    }
+}
+
+impl<const N: usize> PartialEq<[u8; N]> for UnhashedArray<N> {
+    fn eq(&self, other: &[u8; N]) -> bool {
+        &self.inner == other
+    }
+}
+
+/// Translators for keys that are not the length of a standard integer.
+#[derive(Clone, Copy)]
+pub struct Cap<const N: usize> {
+    _marker: UnhashedArray<N>,
```

Comment:
nit: use `core::marker::PhantomData` so that `size_of::<Cap<N>> == 0`.

Reply:
done